### DISCUSSION TITLE
Glpi Agent does not update Inventory

### DISCUSSION THREAD

**Discussion by waaalex** [glpi-project/glpi-agent#588] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

Hello,
For some computers, glpi agent does not update inventory.

Here is error in agent log
```
[Mon Jan 22 12:31:44 2024][info] New inventory from PT-329-2024-01-22-11-16-17 for server0
[Mon Jan 22 12:32:09 2024][error] [http client] communication error: 500 Internal Server Error, Column 'items_id' cannot be null
[Mon Jan 22 12:32:09 2024][error] [http client] server error: Column 'items_id' cannot be null
[Mon Jan 22 12:32:11 2024][info] target server0: next run: Tue Jan 23 12:30:24 2024 - https://glpi/plugins/glpiinventory/
[Mon Jan 22 12:32:11 2024][info] GLPI Agent memory usage: WSS=3469312 PFU=171839488
```

When i do an inventory, agent in glpiinventory is updated but not the computer.
There is an items_id but it is in found as péripheral.
When i search in péripheral for this computer, i do not find it.
It seems like there is a mess in database but there was nothing changed.
I do not know how many computers ar impacted.

### To reproduce

Do an inventory.

### Expected behavior

Computer updated with inventory

### Operating system

Linux

### GLPI Agent version

1.7, 1.6.1

### GLPI version

10.0.11

### GLPIInventory plugin or FusionInventory for GLPI plugin version

GLPI Inventory v1.3.4

**Comment by g-bougard** [comment]
Hi @waaalex 
the error message you're sharing is only telling us there's an issue server-side. The agent seems not the origin of your problem. It just providing datas which triggers a conflict somewhere.
Please, check the GLPI php-errors.log file to find which table seems involved when glpi tries to use a `null` items_id.
In the case you're using a plugin like datainjection, this may make troubles. Try to disable it to see if this helps.

**Comment by waaalex** [comment]
Hello,
Thanks for your answer.
I am using glpiinventory plusgin.
When i disable it, i cannot do an inventory.

When i do an inventory for machine which is impacted, i have nothin in php-error.log.

**Reply by g-bougard** [reply]
If you disable glpiinventory, you have to set a different url for the agent. Check the caution there: https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

What happens when you try to import manually from Administration->Inventory a related computer inventory ?

What other plugins do you have ?

**Comment by waaalex** [comment]
Tested with native inventory and same error.
I think my Database is messed up.

My other plugins : 
fields	
accounts
Form Creator
genericobject
pdf	
reports

**Comment by waaalex** [comment]
Hello,
I think i will restart from scratch.
Is there a procedure to dump/load glpi tickets table with thier dependances?
Thanks.
